src/components/scenes/SendScene.tsx

```diff
@@ -575,21 +576,32 @@ class SendComponent extends React.PureComponent<Props, State> {
     } = this.props
     const { recipientAddress, selectedWalletId } = this.state
     const edgeWallet = currencyWallets[selectedWalletId]
-    const { uniqueIdentifierInfo } = getSpecialCurrencyInfo(edgeWallet.currencyInfo.pluginId)
 
-    if (recipientAddress && uniqueIdentifierInfo != null) {
-      const { addButtonText, identifierName, keyboardType } = uniqueIdentifierInfo
+    const [memoOption] = edgeWallet?.currencyInfo.memoOptions ?? []
+    if (recipientAddress && memoOption != null) {
+      const memoLabel = getMemoLabel(memoOption.memoName)
+      const memoTitle = getMemoTitle(memoOption.memoName)
+      const addButtonText = sprintf(lstrings.unique_identifier_dropdown_option_s, memoLabel)
+
+      let maxLength: number | undefined
+      if (memoOption.type === 'text') {
+        maxLength = memoOption.maxLength
+      } else if (memoOption.type === 'number') {
+        maxLength = memoOption.maxValue?.length
```

Comment:
This is a very rough estimate on the `maxValue` restriction. Line 609 should at least validate the true `maxValue` vs the user input value. Same should be done with hex `maxLength` if inputs could include `0x`

Reply:
Same comment for SendScene2